### ISSUE TITLE
Fail to merge REF and SNP type

### ISSUE THREAD

**Issue by shiquan** [samtools/bcftools#296]
File A:
> chrX  118604436  .  T  .  .  .  .  GT  0

File B:
> chrX  118604436  .  T  A  .  .  .  GT  0/1

Try to use bcftools-1.2 merge (use ` -m both` in default) these files, and the result is 
>
chrX  118604436  .  T  .  .  .  .  GT  0  .
chrX  118604436  .  T  A  .  .  .  GT  ./.  0/1

But clearly we want the result is like this.
>chrX  118604436  .  T  A  .  .  .  GT  0  0/1

I checked  `vcfmerge.c`, and this bug may be fixed by

```
1749  else if ( line_type==VCF_REF ) compatible = 1;   // REF can go with anything
 +  else if ( var_type==VCF_REF ) compatible = 1;  // REF may come first
1750  else if ( var_type&VCF_SNP && line_type&VCF_SNP ) compatible = 1;
```

Thanks for your time.

**Comment by pd3** [comment]
Hi, thank you. This has been recently fixed together with other related issues. It may take some time before the fix appears in the main repository, the fixed version can be in the meantime accessed from here:
http://pd3.github.io/bcftools/
https://github.com/pd3/bcftools/

**Comment by mcshane** [comment]
@pd3's merge changes will be brought in after 1.3 has been released as it is quite a large change.

**Comment by mcshane** [comment]
This should be resolved not that #455 has been merged.
